lib/httpHeaders.ts

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+/**
+ * A collection of HttpHeaders that can be sent with a HTTP request.
+ */
+function getHeaderKey(headerName: string) {
+    return headerName.toLowerCase();
+}
+
+/**
+ * An individual header within a HttpHeaders collection.
+ */
+export interface HttpHeader {
+    /**
+     * The name of the header.
+     */
+    name: string;
+
+    /**
+     * The value of the header.
+     */
+    value: string;
```

Comment:
I feel that saying `value: any` doesn't help consumers of this interface at all. I'd rather make everything a string and let the consumer convert it to a number if they know it will be a number.